contracts/ragnarok/types/BIP.sol

```diff
@@ -0,0 +1,14 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.10;
+
+type BIP is uint256;
+
+library BIPMath {
+    function add(BIP a, BIP b) internal pure returns (BIP c) {
+        c = BIP.wrap(BIP.unwrap(a) + BIP.unwrap(b));
+    }
+
+    function mul(BIP a, uint256 b) internal pure returns (uint256 c) {
```

Comment:
I worry about having division in multiplication that is totally not obvious. This may lead to precision loss. Maybe it should return BP as well and at the and you are should be forced to call something like `normalise` that will divide by 10000?